dashboard/app/models/census/state_cs_offering.rb

```diff
@@ -1490,26 +1506,52 @@ def self.construct_object_key(state_code, school_year, update = 1)
     "state_cs_offerings/#{state_code}/#{school_year}-#{school_year + 1}#{update_string}.csv"
   end
 
-  def self.seed_from_s3
+  def self.seed_from_s3(dry_run = false)
     # State CS Offering data files in S3 are named
     # "state_cs_offerings/<STATE_CODE>/<SCHOOL_YEAR_START>-<SCHOOL_YEAR_END>.csv"
     # The first school year where we have data is 2015-2016
+    seeded_objects = []
     current_year = Date.today.year
     (2015..current_year).each do |school_year|
       SUPPORTED_STATES.each do |state_code|
         SUPPORTED_UPDATES.each do |update|
           object_key = construct_object_key(state_code, school_year, update)
           begin
-            AWS::S3.seed_from_file(CENSUS_BUCKET_NAME, object_key) do |filename|
-              seed_from_csv(state_code, school_year, update, filename)
+            AWS::S3.seed_from_file(CENSUS_BUCKET_NAME, object_key, dry_run) do |filename|
+              seed_from_csv(state_code, school_year, update, filename, dry_run)
+              seeded_objects << object_key
             end
           rescue Aws::S3::Errors::NotFound
             # We don't expect every school year to be there so skip anything that isn't found.
-            CDO.log.warn "State CS Offering seeding: object #{object_key} not found in S3 - skipping."
+            # Note: Don't print out this warning in a dry run to reduce noises.
+            CDO.log.warn "State CS Offering seeding: object #{object_key} not found in S3 - skipping." unless dry_run
           end
         end
       end
     end
+    CDO.log.info "Seeded data from #{seeded_objects.count} object(s)."
+    CDO.log.info seeded_objects.join("\n")
+    CDO.log.info "This is a dry run. No data is written to the database." if dry_run
+  end
+
+  # Test seeding an object from S3 to find issues.
+  # This method does not check if the object had been seeded before
+  # and does not write to the database.
+  # @example:
+  #   Census::StateCsOffering.seed_from_s3_test('AL', 2019)
+  #     will seed from state_cs_offerings/AL/2019-2020.csv object
+  #   Census::StateCsOffering.seed_from_s3_test('WA', 2019, 'x')
+  #     will seed from state_cs_offerings/WA/2019-2020.x.csv object
+  def self.seed_from_s3_test(state_code, school_year, update = 1)
```

Comment:
Getting kind of caught up in the distinction between `dry_run` and `seed_from_s3_test`. I think some confusion coming from the fact that `seed_from_s3` doesn't take the same (or similar) arguments as `seed_from_s3_test`. Maybe `seed_from_s3_test` could be `_test` or something?

Also looks very similar to the inside of the `seed_from_s3` method, wonder if you could dedupe? I see one diff is using `process_file` instead of `seed_from_file`, is that the only difference?

Reply:
I'll change the method name to `dry_seed_single_s3_object` to avoid the confusion. This method can seed one file given to it via the input parameters, while `seed_from_s3` can seed multiple files using a lookup pattern.

The logic in this method is very short and close to the middle part of `seed_from_s3`. They differ in a key way that `AWS::S3.process_file` doesn't check if an object was previously seeded or not (and it also throws a different exception as I just find out). I think it's okay to keep them separate since combining them likely cause more confusion.